Sources/web3swift/Tokens/ERC1376/Web3+ERC1376.swift

```diff
@@ -66,20 +66,15 @@ protocol IERC1376: IERC20 {
 }
 
 // FIXME: Rewrite this to CodableTransaction
-public class ERC1376: IERC1376, ERC20BaseProperties {
-
-    internal var _name: String?
-    internal var _symbol: String?
-    internal var _decimals: UInt8?
-    internal var _hasReadProperties: Bool = false
-
+public class ERC1376: IERC1376, ERCBaseProperties {
+    public var basePropertiesProvder: ERCBasePropertiesProvider
     public var transaction: CodableTransaction
     public var web3: Web3
     public var provider: Web3Provider
     public var address: EthereumAddress
     public var abi: String
```

Comment:
_Food for thought_
Now that I've taken another round looking at ERCs implementations I have a question to which I do not have an answer yet: what is the use case for all these properties being `var` instead of `let`?

It's not like we set `nil` values during the initialization as we just cannot do that.

Making them all `let` is truly a breaking change that is not a one-line fix for all of the users of this library.
I'd make it one of the 4.0.0 goals to make these fields constant using `let`. Unless we will have reasons not to do so and leave them be.

Reply:
Good point. In general, all these classes should probably be checked for improvements again for 4.0.0. I have the feeling there is more to do here.